app/tests/spec/views/confirm_account_unlock.js

```diff
@@ -0,0 +1,258 @@
+      view = new View({
+        router: routerMock,
+        window: windowMock,
+        metrics: metrics,
+        fxaClient: fxaClient,
+        relier: relier,
+        broker: broker,
+        user: user,
+        ephemeralMessages: ephemeralMessages,
+        screenName: 'confirm-account-unlock'
+      });
+      view.VERIFICATION_POLL_IN_MS = 10;
+    }
+
+    beforeEach(function () {
+      routerMock = new RouterMock();
+      windowMock = new WindowMock();
+
+      metrics = new Metrics();
+      relier = new Relier();
+      fxaClient = new FxaClient();
+
+      broker = new OAuthBroker({
+        session: Session,
+        window: windowMock,
+        relier: relier
+      });
+      user = new User();
+
+      account = user.initAccount({
+        email: EMAIL
+      });
+
+      ephemeralMessages = new EphemeralMessages();
+      ephemeralMessages.set('data', {
+        lockoutSource: 'signin',
+        account: account
+      });
+
+      initView();
+
+      return view.render()
+        .then(function () {
+          $('#container').html(view.el);
+        });
+    });
+
+    afterEach(function () {
+      metrics.destroy();
+
+      view.remove();
+      view.destroy();
+
+      view = metrics = null;
+    });
+
+    describe('beforeRender', function () {
+      it('redirects users who browse directly to the page to /signup', function () {
+        sinon.stub(account, 'isEmpty', function () {
+          return true;
+        });
+
+        return view.beforeRender()
+          .then(function () {
+            assert.equal(routerMock.page, 'signup');
+          });
+      });
+    });
+
+    describe('render', function () {
+      it('shows the confirm-account-unlock screen', function () {
+        assert.ok($('#fxa-confirm-account-unlock-header').length);
+      });
+    });
+
+    describe('afterVisible', function () {
+      it('polls', function () {
+        var count = 0;
+        sinon.stub(fxaClient, 'signIn', function () {
+          if (count === 3) {
+            return p({
+              sessionToken: 'newSessionToken'
+            });
+          }
+
+          count++;
+          return p.reject(AuthErrors.toError('ACCOUNT_LOCKED'));
+        });
+
+        // we want the poll to happen.
+        windowMock.setTimeout = window.setTimeout.bind(window);
+
+        return view.afterVisible()
+          .then(function () {
+            assert.equal(fxaClient.signIn.callCount, 4);
+          });
+      });
+
+      it('if caused by signin, notifies the broker when complete', function () {
+        sinon.stub(fxaClient, 'signIn', function () {
+          return p({
+            sessionToken: 'newSessionToken'
+          });
+        });
+
+        sinon.spy(broker, 'afterSignIn');
+
+        return view.afterVisible()
+          .then(function () {
+            assert.isTrue(broker.afterSignIn.called);
+          });
+      });
+
+      it('if not caused by signin, sets the ephemeral messages and goes back', function () {
```

Comment:
"the ephemeral messages" plural?  If so, we should test for the expected number.